src/main/java/com/minecolonies/coremod/commands/ColonyTPCommand.java

```diff
@@ -0,0 +1,240 @@
+            return;
+        }
+
+        EntityPlayer playerToTeleport = null;
+
+        if(sender instanceof EntityPlayer)
+        {
+            playerToTeleport = (EntityPlayer) sender;
+        }
+
+        //If the arguments aren't empty, the sender probably wants to teleport another player.
+        if (args.length != 0 && isPlayerOpped(sender, "ctp"))
+        {
+            final World world = FMLCommonHandler.instance().getMinecraftServerInstance().getEntityWorld();
+            playerToTeleport =
+                    ServerUtils.getPlayerFromUUID(FMLCommonHandler.instance().getMinecraftServerInstance().getPlayerProfileCache()
+                            .getGameProfileForUsername(args[0]).getId(), world);
+
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString("TPin Player: "+ playerToTeleport.getName()));
+        }
+
+        if(playerToTeleport == null)
+        {
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString(CANT_FIND_PLAYER));
+            return;
+        }
+
+        playerToTeleport.getCommandSenderEntity().addChatMessage(new TextComponentString("Buckle up buttercup, this ain't no joy ride!!!"));
+        teleportPlayer(sender, playerToTeleport);
+    }
+
+    /**
+     * Method used to teleport the player.
+     * @param sender the sender to have access to the world.
+     * @param playerToTeleport the player which shall be teleported.
+     */
+    private void teleportPlayer(final ICommandSender sender, final EntityPlayer playerToTeleport)
+    {
+        //Now the position will be calculated, we will try up to 4 times to find a save position.
+        int attCounter = 0;
+        while (attCounter <= ATTEMPTS)
+        {
+            attCounter++;
+            int x = rnd.nextInt(UPPER_BOUNDS) - LOWER_BOUNDS;
+            int z = rnd.nextInt(UPPER_BOUNDS) - LOWER_BOUNDS;
+
+            //Search for a ground position
+            final BlockPos groundPosition = findLand(new BlockPos(x, STARTING_Y, z), sender.getEntityWorld());
+
+            //If no position found
+            if (groundPosition == null)
+            {
+                attCounter++;
+                continue;
+            }
+
+            boolean foundPosition = isPositionSafe(sender, groundPosition) && !findColony(groundPosition, sender.getEntityWorld());
+
+            /* Take the return and determine if good or bad */
+             /* send info to look to see if another colony is near */
+
+            if (foundPosition)
+            {
+                /* everything checks out good make the TP and jump out*/
+                playerToTeleport.setPositionAndUpdate(groundPosition.getX(), groundPosition.getY() + SAFETY_DROP, groundPosition.getZ());
+                return;
+            }
+        }
+        playerToTeleport.getCommandSenderEntity().addChatMessage(new TextComponentString("Couldn't find a safe spot.  Try again in a moment."));
```

Comment:
maybe send this to the sender? because the sender is the one who issued the command and has to reissue it.

maybe return false if it just didnt work